### DISCUSSION TITLE
How to solve EJBCA supporting ED25519 algorithm through PKCS#11 interface?

### DISCUSSION THREAD

**Discussion by vanlee90** [Keyfactor/ejbca-ce#733] [General]
Error: java.lang.IllegalStateException: Algorithm Ed25519 was not recognized

**Comment by primetomas** [comment]
This can not be solved with Sun/Java PKCS#11 because it does not support Ed25519. 

P11NG, an Enterprise feature supports Ed25519, for some HSMs that have support for it. Ed25519 support in HSMs differs between different HSMs so it's not a standard feature.
